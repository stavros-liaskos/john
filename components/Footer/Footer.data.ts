import {FooterI18n} from "./Footer.types";

export const footerI18n: FooterI18n = {
    powered: "Powered by"
}