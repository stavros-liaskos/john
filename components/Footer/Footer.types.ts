export interface FooterI18n {
    powered: string;
}

export interface FooterProps {
    i18n: FooterI18n;
    className: string;
}
