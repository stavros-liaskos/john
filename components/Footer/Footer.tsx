import { FooterProps } from './Footer.types';
import Image from 'next/image';

const Footer = ({ i18n, className }: FooterProps) => {
  if (!i18n || !i18n.powered) {
    return null;
  }

  return (
    <footer className={`border border-indigo-600 ${className}`}>
      <a
        data-testid="footer-powered"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        target="_blank"
        rel="noopener noreferrer"
      >
        {i18n.powered}
        <Image src="/vercel.svg" alt="Vercel Logo" width={72} height={16} />
      </a>
    </footer>
  );
};

export default Footer;
