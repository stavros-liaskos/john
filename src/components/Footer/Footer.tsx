import { FooterProps } from './Footer.types';
import React from 'react';
import Github from '../Icons/github';

const Footer: React.FunctionComponent<FooterProps> = ({ i18n }) => {
  if (!i18n || !i18n.powered) {
    return null;
  }

  return (
    <footer className="rr-column flex-none h-16 border-t-2 rr-border justify-center">
      <a
        className="flex items-center rr-text"
        href="https://github.com/jaivalis/release-raccoon"
        target="_blank"
        rel="noopener noreferrer"
      >
        {i18n.powered}
        <div className="mx-2">
          <Github />
        </div>
      </a>
    </footer>
  );
};
Footer.whyDidYouRender = true;
export default Footer;
