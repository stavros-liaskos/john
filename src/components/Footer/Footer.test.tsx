import React from 'react';
import { getByText, render } from '@testing-library/react';
import Footer from './Footer';
import { footerI18n } from './Footer.data';

describe('Footer', () => {
  it('renders without data without crashing', () => {
    render(<Footer />);
  });

  it('renders with data', () => {
    const { container } = render(<Footer i18n={footerI18n} />);
    const footerPowered = getByText(container, 'Powered by');

    expect(footerPowered).toContainHTML(footerI18n.powered);
    expect(container).toMatchSnapshot();
  });
});
