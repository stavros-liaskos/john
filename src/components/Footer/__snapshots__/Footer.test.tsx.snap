// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders with data 1`] = `
<div>
  <footer
    class="rr-column flex-none h-16 border-t-2 rr-border justify-center"
  >
    <a
      class="flex items-center rr-text"
      href="https://github.com/jaivalis/release-raccoon"
      rel="noopener noreferrer"
      target="_blank"
    >
      Powered by
      <div
        class="mx-2"
      >
        <svg
          class="h-8 w-8 icon-colour"
          fill="none"
          height="24"
          role="img"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M0 0h24v24H0z"
            stroke="none"
          />
          <path
            d="M9 19c-4.286 1.35-4.286-2.55-6-3m12 5v-3.5c0-1 .099-1.405-.5-2 2.791-.3 5.5-1.366 5.5-6.04a4.567 4.567 0 00-1.333-3.21 4.192 4.192 0 00-.08-3.227s-1.05-.3-3.476 1.267a12.334 12.334 0 00-6.222 0C6.462 2.723 5.413 3.023 5.413 3.023a4.192 4.192 0 00-.08 3.227A4.566 4.566 0 004 9.486c0 4.64 2.709 5.68 5.5 6.014-.591.589-.56 1.183-.5 2V21"
          />
        </svg>
      </div>
    </a>
  </footer>
</div>
`;
